crates/sui-types/src/programmable_transaction_builder.rs

```diff
@@ -139,7 +139,13 @@ impl ProgrammableTransactionBuilder {
         Argument::Result(i as u16)
     }
 
-    /// Will fail to generate if given an empty ObjVec
+    /// Insert a new move call into the Programmable Transaction Block.
```

Comment:
...to be honest, it's not clear to me why this function exists.  I think we should probably get rid of it in favour of a single `move_call` helper?  Thoughts @stefan-mysten , @tnowacki ?

Reply:
I am glad you asked this - it's also a question I had.

On that note, can I also ask what is the difference between these two functions, and
1. `crates/sui-transaction-builder/src/lib.rs` in `TransactionBuilder::move_call`
2. also in `crates/sui-transaction-builder/src/lib.rs`, `TransactionBuilder::single_move_call`